pkg/apis/zookeeper/v1beta1/zookeepercluster_types.go

```diff
@@ -157,18 +164,24 @@ func (s *ZookeeperClusterSpec) withDefaults(z *ZookeeperCluster) (changed bool)
 	if s.Pod.withDefaults(z) {
 		changed = true
 	}
-	if s.Persistence == nil {
+	if s.Persistence == nil && !strings.EqualFold(s.StorageType, "ephemeral") {
+		s.StorageType = "persistence"
 		s.Persistence = &Persistence{}
 		changed = true
 	}
-	if s.Persistence.withDefaults() {
+	if strings.EqualFold(s.StorageType, "ephemeral") && s.Ephemeral == nil {
+		s.Ephemeral = &Ephemeral{}
+		s.Ephemeral.EmptyDirVolumeSource = v1.EmptyDirVolumeSource{}
+		changed = true
+	}
+	if !strings.EqualFold(s.StorageType, "ephemeral") && s.Persistence != nil && s.Persistence.withDefaults() {
```

Comment:
Please replace this "if" with "else" for the earlier "if"

Reply:
we can't do that as if the user specifies storageType as ephemeral but gives value for ephemeral structure, in that case, we will end up setting the storageType as persistence.

Reply:
I have restructured this part of the code for readability.